### DISCUSSION TITLE
Include subsection of external file?

### DISCUSSION THREAD

**Discussion by pusolito** [code-hike/codehike#309] [Q&A]
Is it possible to include only a portion of an external file using the `from` annotation?

**Comment by pomber** [comment]
Not possible but it could be a good feature. Could you share more details on why you need it?

**Reply by pusolito** [reply]
This could be useful for Kotlin code, where the list of imports can be really long and not as important for a snippet. Or when there are multiple functions in a file, but the example only needs to refer to a subset of them.

I know `focus` can help here; but it would still include a bunch of noise in the snippet, when a sub-range would be better.

This is something you can do in [Python Markdown](https://facelessuser.github.io/pymdown-extensions/extensions/snippets/) extensions.

**Comment by ZYJLiu** [comment]
Have there been updates to codehike that enable something like this? Currently trying to have codehike reference a single file using the `from` annotation, while providing explanations and only displaying smaller code snippets per step. 

Would the current solution just be to create code snippets that progressively includes more code?

**Reply by pomber** [reply]
will ship it soon #376

**Reply by ZYJLiu** [reply]
Thanks for the update! The Code Hike doc mentions "you can pass the starting and ending line number after the file path". Is it possible to use the new feature to reference multiple sections of another file?

For example, something like:
`# from ./assets/my-file.py 2:3, 5:7, 10`

In order to only show specific lines from the referenced file

**Reply by pomber** [reply]
not possible

**Reply by ZYJLiu** [reply]
Is this something that could be possible in the future? Or is it not planned for the near future because of the comment mentioned in the code folding discussion [here](https://github.com/code-hike/codehike/discussions/310)?

**Reply by pomber** [reply]
It's something that can be added in the future, as part of loading multiple files instead of just one:
````mdx
```js
// from ./foo.js 3:4
// from ./bar.js 3:4
```
````
For your use case, it will be possible to reference the same file multiple times.

If you want, you can create an issue for easier tracking.
